### PR TITLE
Add meta registrations and distributed decomps: _foreach_div_.Scalar, sqrt_.default

### PR DESCRIPTION
**Pull request by janeyx99**
This PR unblocks #104780 by resolving spmd tracing test issues and by adding meta registrations for foreach inplace ops (div_ and sqrt_)

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #104780
* __->__ #104779
* #104669

### CODE REVIEW

File: torch/distributed/_spmd/api.py [hunk 1]
```diff
@@ -271,6 +271,9 @@ def _fused_adam_decomp(
     aten._foreach_mul_.Scalar: partial(
         _foreach_binop_scalar_decomp, aten._foreach_mul.Scalar
     ),
```
Comment (awgu) [inline torch/distributed/_spmd/api.py:273]: Are we actively developing SPMD? 😮

Comment (janeyx99) [reply torch/distributed/_spmd/api.py:273]: Unsure but I had a distributed test failure when I updated foreach Adam to use inplace div and this should fix.

If this is no longer needed, we should remove the tests 😛 and the code 😛 which I'm thinking would be a bigger more involved change than this.

Comment (fegin) [reply torch/distributed/_spmd/api.py:273]: Let me disable the tests.

**Review by fegin — APPROVED** [review]
LGTM, I'll disable _spmd tests.

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**